### PR TITLE
Feature/watch an invoice

### PR DESCRIPTION
**Pull request by nicolasbarb**

### CODE REVIEW

File: easyselling/Scenario/InvoicesScenario/DefaultInvoiceDownloader.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        self.apiCaller = apiCaller
+    }
+
+    private var requestGenerator: AuthorizedRequestGenerator
+    private var apiCaller: APICaller
+
+    func downloadInvoiceFile(id: String, ofType type: String) async throws -> UIImage {
+        let downloadFileRequest = try await requestGenerator.generateRequest(endpoint: .downloadFile,
+                                                                             method: .GET,
+                                                                             headers: ["Content-Type": type],
```
Comment (MaxenceMottard) [inline easyselling/Scenario/InvoicesScenario/DefaultInvoiceDownloader.swift:28]: Remove this, it's the type of the body, not the result type

File: easyselling/Scenario/InvoicesScenario/DefaultInvoiceDownloader.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        self.apiCaller = apiCaller
+    }
+
+    private var requestGenerator: AuthorizedRequestGenerator
+    private var apiCaller: APICaller
+
+    func getInvoiceFile(of fileId: String) async throws -> InvoiceFile {
+        let fileRequest = try await requestGenerator.generateRequest(endpoint: .fileById,
+                                                                 method: .GET,
+                                                                 headers: ["Content-Type": "application/json"],
```
Comment (MaxenceMottard) [inline easyselling/Scenario/InvoicesScenario/DefaultInvoiceDownloader.swift:52]: Remove, it's automatically added by this class: https://github.com/ynov-lyon-mobilite/easyselling-ios/blob/develop/easyselling/Services/DefaultRequestGenerator.swift

File: easyselling/Scenario/InvoicesScenario/InvoiceViewer/InvoiceView.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+//
+
+import SwiftUI
+
+struct InvoiceView: View {
+
+    @ObservedObject var viewModel: InvoiceViewerViewModel
+
+    var body: some View {
+        VStack {
```
Comment (MaxenceMottard) [inline easyselling/Scenario/InvoicesScenario/InvoiceViewer/InvoiceView.swift:15]: Why use this useless VStack with only one child ?

File: easyselling/Scenario/InvoicesScenario/InvoiceViewer/InvoiceViewerViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+import Foundation
+
+class InvoiceViewerViewModel: ObservableObject {
+
+    init(invoiceFile: File) {
+        self.invoiceFile = invoiceFile
+    }
+
+    @Published var invoiceFile: File
```
Comment (MaxenceMottard) [inline easyselling/Scenario/InvoicesScenario/InvoiceViewer/InvoiceViewerViewModel.swift:16]: Do you really need **@Published**? For me this should be a **let** variable

File: easyselling/Scenario/InvoicesScenario/VehicleInvoicesView.swift [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+    var body: some View {
+        VStack {
+            if viewModel.isLoading {
+                ProgressView()
+            } else {
+                List(viewModel.invoices) { invoice in
+                    VStack {
+                        if viewModel.isDownloading && viewModel.chosenInvoice == invoice.id {
+                            HStack {
```
Comment (MaxenceMottard) [inline easyselling/Scenario/InvoicesScenario/VehicleInvoicesView.swift:22]: Useless

File: easyselling/Services/DefaultAPICaller.swift [hunk 6]
```diff
@@ -57,4 +61,33 @@ final class DefaultAPICaller: APICaller {
             throw APICallerError.from(statusCode: strongResponse.statusCode)
         }
     }
+
+    func callImage(_ urlRequest: URLRequest) async throws -> UIImage {
```
Comment (MaxenceMottard) [inline easyselling/Services/DefaultAPICaller.swift:65]: This should not appear in api caller. It's only used by one service so it shouldn't be there.

Comment (MaxenceMottard) [reply easyselling/Services/DefaultAPICaller.swift:65]: Like this : https://github.com/ynov-lyon-mobilite/easyselling-ios/blob/develop/easyselling/Services/DefaultFileUploader.swift

Comment (MaxenceMottard) [reply easyselling/Services/DefaultAPICaller.swift:65]: Moreover, it is not the role of the service to convert data into UI

**Review by MaxenceMottard — CHANGES_REQUESTED** [review]

File: easyselling/Scenario/InvoicesScenario/InvoiceViewer/InvoiceView.swift (hunk 3 above)
Comment (nicolasbarb) [reply easyselling/Scenario/InvoicesScenario/InvoiceViewer/InvoiceView.swift:15]: Pour plus tard

File: easyselling/Services/DefaultAPICaller.swift [hunk 7]
```diff
@@ -37,6 +38,7 @@ final class DefaultAPICaller: APICaller {
             }
 
             do {
+                print(data)
```
Comment (MaxenceMottard) [inline easyselling/Services/DefaultAPICaller.swift:41]: Remove

File: easysellingTests/DefaultAPICaller_Specs/FailingAPICaller.swift [hunk 8]
```diff
@@ -7,6 +7,8 @@
 
 import Foundation
 import Combine
+import UIKit
```
Comment (MaxenceMottard) [inline easysellingTests/DefaultAPICaller_Specs/FailingAPICaller.swift:10]: Remove

File: easyselling/Scenario/InvoicesScenario/InvoiceViewer/InvoiceView.swift (hunk 3 above)
Comment (MaxenceMottard) [inline easyselling/Scenario/InvoicesScenario/InvoiceViewer/InvoiceView.swift:15]: Still useless

**Review by MaxenceMottard — CHANGES_REQUESTED** [review]

**Review by MaxenceMottard — APPROVED** [review]

**Pull request merged**